python_modules/dagster-graphql/dagster_graphql/schema/pipelines/resource.py

```diff
@@ -6,17 +6,26 @@
 from ..config_types import GrapheneConfigTypeField
 
 
+class GrapheneResourceSource(graphene.Enum):
+    FROM_OVERRIDE = "FROM_OVERRIDE"
+    FROM_DEFAULT = "FROM_DEFAULT"
+
+    class Meta:
+        name = "ResourceSource"
+
+
 class GrapheneResource(graphene.ObjectType):
     name = graphene.NonNull(graphene.String)
     description = graphene.String()
     configField = graphene.Field(GrapheneConfigTypeField)
+    source = graphene.Field(GrapheneResourceSource)
```

Comment:
is this always populated?  should this be non-null?

Reply:
not always populated I think